sdk/playwrighttesting/microsoft-playwright-testing/src/utils/serviceClient.ts

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import type { FullResult } from "@playwright/test/reporter";
+import { AxiosResponse } from "axios";
+import Debug from "debug";
+import { backOff } from "../common/backoff";
+import { Constants } from "../common/constants";
+import { EnvironmentVariables } from "../common/environmentVariables";
+import { HttpService } from "../common/httpService";
+import { Shard, UploadMetadata } from "../model/shard";
+import { StorageUri } from "../model/storageUri";
+import { TestResult } from "../model/testResult";
+import { TestRun } from "../model/testRun";
+import { CIInfo } from "./cIInfoProvider";
+import ReporterUtils from "./reporterUtils";
+
+const debug = Debug(Constants.DEBUG_NAMESPACE);
+
+export class ServiceClient {
```

Comment:
@xirzec I think they can use @azure/core-client here can't they?

Reply:
Probably you could accomplish everything here with `@azure/core-rest-pipeline`

Reply:
moved to `@azure/core-rest-pipeline`